### DISCUSSION TITLE
Allow for seeing past chat

### DISCUSSION THREAD

**Discussion by mtanco** [h2oai/wave#744] [Q&A]
The `chat_card` only shows you information that has been sent while you were in the app, so if you refresh the browser all chat goes away as the Client is re-initialized. Would it make sense to add the ability to specify that you would like users to be able to see past chats?

https://user-images.githubusercontent.com/4793561/114947330-d0ca3200-9e01-11eb-95a5-d7afc777cb40.mov

**Comment by lo5** [comment]
Assuming the chat card was added to the page in Wave app (not script) -

The past messages are currently tied to the page. In case of chat cards added in an app with mode `unicast`, the page ("what you see in the UI") is different for different users, hence the messages are not carried over. 

To allow the chat messages to be retained, either the app mode needs to be `multicast` (or "global to all users") OR the chat card needs to be created on a standalone page (via a Wave script).

There is currently no way (or API) for an app to capture the chat stream.

**Comment by mtanco** [comment]
> To allow the chat messages to be retained, either the app mode needs to be multicast (or "global to all users") OR the chat card needs to be created on a standalone page (via a Wave script).

This is interesting, I will put together an example.
